packages/integrations/src/offline.ts

```diff
@@ -0,0 +1,101 @@
+import { captureEvent } from '@sentry/minimal';
+import { Event, EventProcessor, Hub, Integration } from '@sentry/types';
+import { getGlobalObject, logger, uuid4 } from '@sentry/utils';
+// @ts-ignore
+import localforage = require('localforage');
+
+/**
+ * cache offline errors and send when connected
+ */
+export class Offline implements Integration {
+  // tslint:disable: no-unsafe-any
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Offline';
+
+  /**
+   * @inheritDoc
+   */
+  public readonly name: string = Offline.id;
+
+  /**
+   * event cache
+   */
+  public offlineEventStore: any;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor() {
+    this.offlineEventStore = localforage.createInstance({
+      name: 'sentry/offlineEventStore',
+    });
+
+    getGlobalObject<Window>().addEventListener('online', () => {
+      this._sendEvents().catch(() => {
+        logger.warn('could not send cached events');
+      });
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(addGlobalEventProcessor: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    addGlobalEventProcessor(async (event: Event) => {
```

Comment:
We try to avoid using `async` in non-test codebase as it can have much higher bundle size vs. just using regular Promises.

Reply:
Ah, got it. I can convert it.

Reply:
Okay, I removed the `async/await` calls.